src/test/java/org/cactoos/iterable/MatchedTest.java

```diff
@@ -155,4 +157,40 @@ public void iterablesOfDifferentTypes() {
         ).affirm();
     }
 
+    @Test
+    public void traversableSeveralTimes() {
+        final Iterable<Integer> matched = new Matched<>(
+            (Number fst, String snd) -> fst.intValue() == snd.length(),
+            new RangeOf<>(1, 3, value -> value + 1),
+            new IterableOf<>("1", "11", "111")
+        );
+        matched.forEach(Assertions::assertNotNull);
+        matched.forEach(Assertions::assertNotNull);
```

Comment:
@andreoss let's find an alternative way of testing this, we shouldn't be using `org.junit.jupiter.api.Assertions`. Actually this class should have been added to the forbidden-api list (see pom.xml). I will be adding it in another PR.

One solution would be not to bother testing if the content is null, as it is not really related to what this test seems to be testing.